### DISCUSSION TITLE
Questions about docker compose

### DISCUSSION THREAD

**Discussion by 1mfaasj** [Freika/dawarich#468] [Q&A] [answered]
Hi,
I want to deploy dawarich with docker compose and have some questions.
I cannot find the answers on the website.

What I want to deploy:
https://github.com/Freika/dawarich/blob/master/docker-compose_mounted_volumes.yml

can you please explain what the purpose is of these volumes?
I can imagine what the cache does but not the public and watched?
```
      - /dawarich/cache:/usr/local/bundle/gems
      - /dawarich/public:/var/app/dawarich_public
      - /dawarich/watched:/var/app/tmp/imports/watched
```

the database and keydb have this shared volume, why?
`/dawarich/shared:/var/shared`

why are these needed?
```
RAILS_ENV=development

    stdin_open: true
    tty: true
    entrypoint: dev-entrypoint.sh
    command: [ 'bin/dev' ]
```

and the last question: I want to deploy my own PHOTON_API_HOST
do you maybe have a docker compose example for this?

thanks in advance
Kind regards

**Comment by Freika** [comment]
- `/dawarich/cache:/usr/local/bundle/gems` — for caching installed dependencies
- `/dawarich/public:/var/app/dawarich_public` — exports are being written in the public directory
- `/dawarich/watched:/var/app/tmp/imports/watched` — is used by [Watcher](https://dawarich.app/docs/tutorials/watcher)
- `/dawarich/shared:/var/shared` — will have to check, might be a piece of legacy
- `RAILS_ENV=development` — to explicitly provide environment name, as of now, Dawarich is being run in development to make remote debugging easier
- `stdin_open`; `tty` — to allow interactive sessions in terminal, i.e. access and use the [console](https://dawarich.app/docs/FAQ#how-to-enter-dawarich-console)
- `entrypoint`; `command` — to run the app
- own photon instance — https://dawarich.app/docs/tutorials/reverse-geocoding#setting-up-your-own-reverse-geocoding-service

**Comment by 1mfaasj** [comment]
Thanks for the answer! It makes a lot clearer.
about the shared volume, please remove legacy stuff that is no longer in use or has no purpose. that makes deploying dawarich easier and more understandable.

RAILS_ENV=development, 
can I set other ENV's of this later? for example development/acceptance production? or is this something for you as a developer of the solution? if so: then I expect you to do this in a docker tag and not in a docker env.

May I ask why I should need the console? what is the use case behind this? since I can manage it all from the GUI?

On what basis are certain gps locations linked to each other? then you see a blue line on the map.
and can I change this behaviour? that I only see the gps locations on the map, as a dot and don't see any underlying 'connections' between them? because that would suggest that I walked/drove that distance while that could not have been the case.

I see a KM amount for each year/month ... is this based on these GPS 'connections'? if so; the km value could be not correct at all

Other question: I imported the Immich locations points, when I add new photo's to Immich, does Dawarich respond and retrieve these values? ( I like automation )

another suggestion I want to give you: since the development is still going fast; try to get the base robust first before working on new things like 'visits'. I dont really see the added value of that (yet). what i would find much more important is maps and stats. These should be fast and in terms of filtering you can maybe add a few things to make it even more powerful.

**Reply by Freika** [reply, marked as answer]
The other env is `production` although it's not recommended to run app in.

The console might come useful to resolve some edge-case issues that weren't addressed in the code yet.

> On what basis are certain gps locations linked to each other? then you see a blue line on the map.

https://github.com/Freika/dawarich/blob/master/CHANGELOG.md#changed-57

> I see a KM amount for each year/month ... is this based on these GPS 'connections'? if so; the km value could be not correct at all

It's based on distance between consecutive points

> Other question: I imported the Immich locations points, when I add new photo's to Immich, does Dawarich respond and retrieve these values? ( I like automation )

Not yet, it will be automated in the future
